src/utils/wallet.ts

```diff
@@ -1,14 +1,15 @@
 // Set of helper functions to facilitate wallet setup
 
+import { ExternalProvider } from '@ethersproject/providers'
 import { BASE_BSC_SCAN_URL, BASE_URL } from 'config'
 import { nodes } from './getRpcUrl'
 
 /**
  * Prompt the user to add BSC as a network on Metamask, or switch to BSC if the wallet is on a different network
  * @returns {boolean} true if the setup succeeded, false otherwise
  */
-export const setupNetwork = async () => {
-  const provider = window.ethereum
+export const setupNetwork = async (externalProvider?: ExternalProvider) => {
+  const provider = externalProvider || window.ethereum
```

Comment:
blocto, coinbase wallet inject ethereum provider, to prevent multiple wallet installed and we don't know which wallet need to setup, pass web3 react library's provider here to be much more determine